From c0a58d2f201746c60f48465ed1e2debe5eb7bb89 Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Fri, 22 Aug 2014 03:52:07 -0400
Subject: [acpi] ras/extlog: Adjust init flow

Message-id: <1408679528-32070-10-git-send-email-ruwang@redhat.com>
Patchwork-id: 88696
O-Subject: [PATCH RHEL7.1 BZ1085141 09/10] RAS, extlog: Adjust init flow
Bugzilla: 1085141
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: "Chen, Gong" <gong.chen@linux.intel.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1085141

Backport of:
Original commit: 7c76bb5f7a3d052339b873374333dd0dcc35ce28
Original Author: "Chen, Gong" <gong.chen@linux.intel.com>
Original Date: Wed, 11 Jun 2014 04:34:51 -0400

 Unless the platform has eMCA related capability, don't
 need to check if there is conflict with EDAC driver.

Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/acpi_extlog.c b/drivers/acpi/acpi_extlog.c
index ff9e9c8..9b388b1 100644
--- a/drivers/acpi/acpi_extlog.c
+++ b/drivers/acpi/acpi_extlog.c
@@ -260,19 +260,16 @@ static int __init extlog_init(void)
  u64 cap;
  int rc;
 
+ rdmsrl(MSR_IA32_MCG_CAP, cap);
+
+ if (!(cap & MCG_ELOG_P) || !extlog_get_l1addr())
+  return -ENODEV;
+
  if (get_edac_report_status() == EDAC_REPORTING_FORCE) {
   pr_warn("Not loading eMCA, error reporting force-enabled through EDAC.\n");
   return -EPERM;
  }
 
- rc = -ENODEV;
- rdmsrl(MSR_IA32_MCG_CAP, cap);
- if (!(cap & MCG_ELOG_P))
-  return rc;
-
- if (!extlog_get_l1addr())
-  return rc;
-
  rc = -EINVAL;
  /* get L1 header to fetch necessary information */
  l1_hdr_size = sizeof(struct extlog_l1_head);
-- 
1.7.1